include/posix/pthread.h

```diff
@@ -7,39 +7,37 @@
 #ifndef __PTHREAD_H__
 #define __PTHREAD_H__
 
-#ifdef CONFIG_NEWLIB_LIBC
+#include <kernel.h>
 #include <time.h>
-#else
-/* This should probably live somewhere else but Zephyr doesn't
- * currently have a stdc layer to provide it
+#include "sys/types.h"
+#include "posix_sched.h"
+#include "unistd.h"
+
+/* Pthread cancellation */
+#define PTHREAD_CANCEL_POS     0
+/* Non Essential Thread */
+#define PTHREAD_CANCELABLE     (0 << PTHREAD_CANCEL_POS)
+/* Essential Thread */
+#define PTHREAD_NON_CANCELABLE (1 << PTHREAD_CANCEL_POS)
+#define K_LOWEST_STACK_SIZE     256
```

Comment:
how was this value arrived at?

Reply:
I just went through the Zephyr Code base and looked that 256 is least stack size used by a thread.

Reply:
Hi @andrewboie Sorry for confusion. But i have mentioned clearly on PR https://github.com/zephyrproject-rtos/zephyr/pull/5494 that discussion for this patch is moved to https://github.com/zephyrproject-rtos/zephyr/pull/5667.
@nashif requested to merge all POSIX related patches into 1 PR and for that I created this PR later.
To avoid confusion, I have closed other POSIX related PRs.

Reply:
You don't need this value at all unless I'm missing something.
It only seems to be used for the defaults which have a NULL stack pointer anyway,
Just use NULL for the default stack pointer and 0 for the size. There is no actual minimum. I have a stack in a double fault handler which is about 12 bytes in size.

Reply:
hmm got it, So basically your suggestion to to define stack_size as 0, something like stack pointer. Let the user provide stack_size along with stack pointer,

Reply:
> Hi @andrewboie Sorry for confusion. But i have mentioned clearly on PR #5494 that discussion for this patch is moved to #5667.

Don't have multiple open PRs with the same code. Just don't do it. Close one of them.